### PR TITLE
Add umpire package

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: var/spack/repos/builtin/packages/umpire/package.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        spec = self.spec
+
+        options = []
+
+        if '+cuda' in spec:
+            options.extend([
+                '-DENABLE_CUDA=On',
+                '-DCUDA_TOOLKIT_ROOT_DIR=%s' % (spec['cuda'].prefix)])
+
+        if '~cuda' in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/umpire/package.py]: you can just do `else:` here.

File: var/spack/repos/builtin/packages/umpire/package.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            options.extend([
```
Comment (davydden) [inline var/spack/repos/builtin/packages/umpire/package.py]: use `append`

File: var/spack/repos/builtin/packages/umpire/package.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                '-DENABLE_CUDA=Off'])
+
+        if '+fortran' in spec:
+            options.extend([
```
Comment (davydden) [inline var/spack/repos/builtin/packages/umpire/package.py]: use `append`

File: var/spack/repos/builtin/packages/umpire/package.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Umpire(CMakePackage):
+    """An application-focused API for memory management on NUMA & GPU
+    architectures"""
+    homepage = "https://github.com/LLNL/Umpire"
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/umpire/package.py]: Can you add a URL for this package? Something like:
```python
url = "https://github.com/LLNL/Umpire/archive/v0.1.4.tar.gz"
```
Spack does weird things when a package doesn't have any URL. In particular, `spack uninstall` is known to break.

Comment (davidbeckingsale) [reply var/spack/repos/builtin/packages/umpire/package.py]: URL added.

**Pull request merged**